### PR TITLE
Add config files from wheel

### LINKED ISSUE #233: Add configuration files from wheel
- [ ] Rename all `js` files with React code with `jsx` extension.
- [ ] Add all the ESlint, and Prettier configuration from [wheel](https://github.com/bigbinary/wheel).
- [ ] Run eslint and prettier on all the files.
```zsh
npx prettier --write "./lib/**/*.{js,jsx,json}"
npx eslint --fix "./lib/**/*.{js,jsx,json}"
npx prettier --write "./example/**/*.{js,jsx,json}"
npx eslint --fix "./example/**/*.{js,jsx,json}"
```
- [ ] Add git hooks using husky, other config files (if any).

Refer: https://github.com/bigbinary/neeto-filters/pull/11 if you faced any blockers.

@mohit-s96 _a please take this up.

### PR DESCRIPTION
**Pull request by mohit-s96**
closes #233

### CODE REVIEW

**Comment by mohit-s96** [comment]
@AbhayVAshokan _a. Please review. There are many changes here, except a few config changes, almost all of them are styling/linting related.

File: package.json [hunk 1]
```diff
@@ -93,26 +109,29 @@
   "scripts": {
     "start": "NODE_ENV=development webpack-dev-server --open --mode development --config webpack.dev.config.js",
     "build": "NODE_ENV=production webpack --config webpack.example.config.js",
-    "webpack-build": "NODE_ENV=production webpack --mode production"
+    "webpack-build": "NODE_ENV=production webpack --mode production",
+    "prepare": "husky install"
+  },
+  "lint-staged": {
+    "{src,example}/**/*.{js,jsx,json}": [
```
Comment (AbhayVAshokan) [inline package.json:116]:
```suggestion
    "{lib,example}/**/*.{js,jsx,json}": [
```

File: babel.config.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      isProductionEnv && [
+        "babel-plugin-transform-react-remove-prop-types",
+        {
+          removeImport: true,
+        },
+      ],
+      isDevelopmentEnv && "react-refresh/babel",
+    ].filter(Boolean),
+  };
+};
```
Comment (AbhayVAshokan) [inline babel.config.js:85]: It is better to restrict the babel config to what is required. If the `.babelrc` file is encountered during rebase (while resolving conflicts), please delete `.babelrc` and keep `babel.config.js` file.

```suggestion
module.exports = api => {
  api.cache(true);

  return {
    presets: ["@babel/preset-env", "@babel/preset-react"],
    plugins: ["@babel/plugin-proposal-class-properties"],
    env: {
      production: {
        plugins: [
          "@babel/plugin-proposal-class-properties",
          "@babel/plugin-transform-runtime",
        ],
      },
    },
  };
};

```

**Review by AbhayVAshokan — CHANGES_REQUESTED** [review]
@mohit-s96 _a please check the changes.

**Comment by mohit-s96** [comment]
@AbhayVAshokan _a. Please review the changes.

**Review by AbhayVAshokan — APPROVED** [review]
LGTM
@mohit-s96 _a please rebase with master to resolve all conflicts and proceed to merge.

**Pull request merged**